fix #285651 swap chord symbol placement with X shortcut

peterhieuvu opened the pull request:
All the infrastructure was already set up, so all that was needed was to add a line to allow the shortcut to apply to chord symbols too.

MarcSabatella commented:
Thanks, was just noticing this was missing!  Indeed, this is all that should be needed.

FWIW, though, right now there is a style setting to control the actual offset when flipped below, but nothing in the UI to set that value.  Eventually we should have position above / below in Format / Style / Chord Symbols, like we do for staff text and some others.

peterhieuvu commented:
Ah that's true. I'll look into that in the future!

peterhieuvu commented:
Made another commit for the suggestion you mentioned. I'm not sure if I was supposed to rebase and force-push, but I ended up doing so. I also modified the default value for position below since, before, it was about 10 away from actually doing anything substantial. I also added reset buttons for the other spin boxes having to do with positioning. 

I used Qt Designer which ended up rearranging the .ui XML which messed with the diff quite a bit and made it look like I made changes that I didn't (the stuff it moved around had to do with similar elements so it was as if I changed the existing elements when the attributes were just the attributes of the new elements that ended up being in the same spot). I don't have much experience with Qt, so I'm not sure if this was the proper way to edit the UI. Let me know! Thanks.

Jojo-Schmitz commented on libmscore/style.cpp:
```diff
@@ -269,10 +269,10 @@ static const StyleType styleTypes[] {
-      { Sid::chordSymbolAPosBelow,      "chordSymbolPosBelow",       QPointF(.0, 3.5) },
+      { Sid::chordSymbolAPosBelow,      "chordSymbolPosBelow",       QPointF(.0, 7.0) },
 
       { Sid::chordSymbolBPosAbove,      "chordSymbolBPosAbove",      QPointF(.0, -5.0) },
```
This is 2.5 more than `Sid::chordSymbolAPosAbove`, which seems OK

Jojo-Schmitz commented on libmscore/style.cpp:
```diff
@@ -269,10 +269,10 @@ static const StyleType styleTypes[] {
 
       { Sid::chordSymbolBPosAbove,      "chordSymbolBPosAbove",      QPointF(.0, -5.0) },
-      { Sid::chordSymbolBPosBelow,      "chordSymbolBPosBelow",      QPointF(.0, 3.5) },
+      { Sid::chordSymbolBPosBelow,      "chordSymbolBPosBelow",      QPointF(.0, 7.0) },
```
but then this should be 2.5 below `Sid::chordSymbolAPosBelow`, shouldn't it?

peterhieuvu replied:
Oh yeah you're right, I will make that change once we decide what's a good offset for position below.

Jojo-Schmitz replied:
I think those B types should be farther away from staff, so above A  type above but below A type below

Jojo-Schmitz replied:
but if the same value for either results in that, like die to autoplace, no change needed, then actually one could use A above and B below without getting a huge gap?

peterhieuvu replied:
Oh yeah that makes sense. Originally, the position below for both A and B were the same so that's why I kept it. I think that's a good reason why.

Jojo-Schmitz commented on libmscore/style.cpp:
```diff
@@ -269,10 +269,10 @@ static const StyleType styleTypes[] {
 
       { Sid::chordSymbolAPosAbove,      "chordSymbolPosAbove",       QPointF(.0, -2.5) },
-      { Sid::chordSymbolAPosBelow,      "chordSymbolPosBelow",       QPointF(.0, 3.5) },
+      { Sid::chordSymbolAPosBelow,      "chordSymbolPosBelow",       QPointF(.0, 7.0) },
```
I thing the old values are OK here

peterhieuvu replied:
I was considering leaving the old value, but the only thing was that this offset is calculated from the top of the staff so 3.5 places the chord symbol inside the staff. Let me know if you have any further reasoning for this though! Thank you for the feedback.

Jojo-Schmitz replied:
of course the default setting should make the chord symbol show below staff.

peterhieuvu replied:
I'll take that meaning that it's okay to keep the value :) The old value works with autoplace on of course, but even then if you want to shift it a bit, you still have to move it past 6.5 for it to change so I think this would still be useful when autoplace is on.

MarcSabatella commented:
From my perspective, it makes sense to have defaults for above the staff that work well for A & B.  I'm not sure there would be any agreement on what actually does make sense for below, as long as A does something reasonable.  Probably most people putting chords under the staff would never use B.  If they did, I could imagine still wanting them higher than A, but that would mean moving A lower than you'd have them otherwise.  Maybe it's actually best to make the B position below the same as A.  Then A is already in the right place if that's all you use, and if you do choose to use B also, you only have to change that (or let autoplace sort things out for you).  Also, this is the probably the best answer if people want to place A above, B below - both A and B will be close to staff.

In fact, I've talked myself into it, that's the right answer :-)

Jojo-Schmitz commented:
Continued in #4993 I guess

peterhieuvu commented:
I closed this one for now because I think this PR is a bit too scattered with what it changed and I don't really agree with the UI stuff that I changed here anymore. I think I'll revisit these in the future with more feedback because I don't really have experience with design and stuff.
